packages/nouns-webapp/src/components/ProposalActionsModal/steps/LumpSumReviewStep/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { Trans } from '@lingui/macro';
+import { utils } from 'ethers';
+import React from 'react';
+import { FinalProposalActionStepProps, ProposalActionModalState } from '../..';
+import ShortAddress from '../../../ShortAddress';
+import { SupportedCurrency } from '../LumpSumDetailsStep';
+import classes from './LumpSumReviewStep.module.css';
+import payerABI from '../../../../utils/payerContractUtils/payerABI.json';
+import ModalBottomButtonRow from '../../../ModalBottomButtonRow';
+import ModalTitle from '../../../ModalTitle';
+import config from '../../../../config';
+
+const handleActionAdd = (state: ProposalActionModalState, onActionAdd: (e?: any) => void) => {
+  if (state.lumpSumCurrency === SupportedCurrency.ETH) {
+    onActionAdd({
+      address: state.address,
+      value: state.amount ? utils.parseEther(state.amount.toString()).toString() : '0',
+      signature: undefined,
+      calldata: '0x',
+    });
+  } else if (state.lumpSumCurrency === SupportedCurrency.USDC) {
+    const signature = 'sendOrRegisterDebt(address,uint256)';
+    const abi = new utils.Interface(payerABI);
+    onActionAdd({
```

Comment:
what about adding a transaction to fund TokenBuyer with ETH?
description in the spec [here](https://verbs.notion.site/Draft-Nouns-Proposal-Builder-v2-Requirements-0008cfe4b6654d6a8028c7c49c163d3a)

Reply:
Added some code here to add / remove a token buyer top up if the set of proposal actions has at least 1 USDC lump sum payment 


asked a question on discord about exact function to call but add / remove mechanism seems to work